cpp/src/arrow/engine/substrait/relation_internal.h

```diff
@@ -38,6 +38,8 @@ struct DeclarationInfo {
 
   /// The number of columns returned by the declaration.
   int num_columns;
+
+  std::shared_ptr<Schema> output_schema;
```

Comment:
I feel like this supersedes `num_columns`; `num_columns` should always equal the number of fields in the schema. I only put that there because I didn't want to have to bother with the whole schema yet.

Reply:
Yes this make sense. I will update this.